app/lib/katello/resources/discovery/file_discovery.rb

```diff
@@ -0,0 +1,34 @@
+module Katello
+  class FileDiscovery < RepoDiscovery
+    attr_reader :found, :crawled, :to_follow
+    def initialize(url, upstream_username = nil,
+                   upstream_password = nil, search = '*', crawled = [],
+                   found = [], to_follow = [])
+      @uri = uri(url)
+      @upstream_username = upstream_username.empty? ? nil : upstream_username
+      @upstream_password = upstream_password.empty? ? nil : upstream_password
+      @search = search
```

Comment:
These parameters are unused. Perhaps drop them? If not, then should the constructor be part of the base class?

Reply:
I will drop these here..
I was trying to get the constructor as part of base class but the foreman task output which is read in chunks doesn't work when inherited for some reason. Had to go with constructors in child classes and the class_for construct.

Reply:
Updated to make credentials and search optional additional params for container and yum classes.